docs/decisions/0003-lti-1p3-score-linking.rst

```diff
@@ -0,0 +1,96 @@
+LTI Advantage AGS Score Linking
+-------------------------------
+
+Status
+======
+
+In Review
+
+Context
+=======
+
+LTI Advantage provides new ways for LTI tools to push grades back into the platform through the `Assignment and Grades Services (AGS)`_,
+which don't map 1:1 with the grading and gradebook structure present in Open edX.
+
+There's two models of interaction to pushing grades to the platform in the LTI AGS services:
+
+1. Declarative: the platform creates a LineItem (equivalent of a gradebook line/grade) and tools can only push results to that item.
+2. Programmatic: the tool uses the AGS endpoints to manage it's own line items and grades. The tool is responsible for linking each line item to the resourceLinks, which means that a tool might not link a grade to it's respective problem.
+
+See a more detailed description in the `Coupled vs decoupled line items`_ section of the spec.
+
+.. _`Assignment and Grades Services (AGS)`: https://www.imsglobal.org/spec/lti-ags/v2p0
+.. _`Coupled vs decoupled line items`: https://www.imsglobal.org/spec/lti-ags/v2p0#coupled-vs-decoupled-line-items
+
+
+Decisions
+=========
+
+We want to enable the platform to have full LTI Advantage compatibility, so we need to allow both interaction models to happen.
+
+To do that, a configuration option will be available to course creators when adding a new graded LTI 1.3 block to a course.
+This configuration will be called *Tool Grading Configuration* and offer the following options:
+
+.. list-table::
+   :widths: auto
+   :header-rows: 1
+
+   * - Option name
+     - Behavior
+   * - Only allow tools to push a single grade per student (Default)
+     - This is the "declarative" approach, where the platform will create a single LineItem, and only allow read-only access to it.
+       The tool is only able to read LineItems, push scores and retrieve grades.
+   * - Allow tools to manage their own grades
```

Comment:
I'm wondering how to help the course author make this decision.   The spec says,
> [programmatic access] requires an elevated level of trust from the platform which might be cautious to which tools it grants POST on 'LineItems.url'.

I imagine tools will indicate that they need this access?  So the instructor instructions could be, "Only use this option if the tool requires it."  Are we worried about risk?  "elevated level of trust" implies that something could go wrong.  I suppose a malicious or badly written tool could send thousands of line items.  But it's very unlikely.